src/private/WRLDRouteView.mm

```diff
@@ -0,0 +1,301 @@
+
+                if (!isValidTransition)
+                {
+                    continue;
+                }
+
+                WRLDRouteStep* stepBefore = [section.steps objectAtIndex:i-1];
+                WRLDRouteStep* stepAfter = [section.steps objectAtIndex:i+1];
+                
+                [self addLineCreationParamsForStep:step stepBefore:stepBefore stepAfter:stepAfter flattenStepIndex:flattenStepIndex color:m_color];
+            }
+            else
+            {
+                [self addLineCreationParamsForStep:step ndFlattenStepIndex:flattenStepIndex];
+            }
+            
+            flattenStepIndex++;
+        }
+    }
+    [self refreshPolylines];
+}
+
+-(void) updateRouteProgress:(int)sectionIndex
+                  stepIndex:(int)stepIndex
+        closestPointOnRoute:(CLLocationCoordinate2D)closestPointOnRoute
+indexOfPathSegmentStartVertex:(int)indexOfPathSegmentStartVertex
+{
+    [self removeFromMap];
+    
+    NSMutableArray* sections = m_route.sections;
+    int flattenStepIndex = 0;
+    
+    for (int j=0; j<sections.count;j++)
+    {
+        WRLDRouteSection* section = [sections objectAtIndex:j];
+        NSMutableArray* steps = section.steps;
+        for (int i=0;i<steps.count;i++)
+        {
+            WRLDRouteStep* step = [steps objectAtIndex:i];
+            if (step.pathCount < 2)
+            {
+                continue;
+            }
+            
+            BOOL isActiveStep = sectionIndex == j && stepIndex == i;
+            
+            if (step.isMultiFloor)
+            {
+                bool isValidTransition = i > 0 && i < (section.steps.count - 1) && step.isIndoors;
+                
+                if (!isValidTransition)
+                {
+                    continue;
+                }
+                
+                WRLDRouteStep* stepBefore = [steps objectAtIndex:i-1];
+                WRLDRouteStep* stepAfter = [steps objectAtIndex:i+1];
+                
+                if(isActiveStep) {
+                    bool hasReachedEnd = indexOfPathSegmentStartVertex == (step.pathCount-1);
+                    
+                    [self addLineCreationParamsForStep:step stepBefore:stepBefore stepAfter:stepAfter flattenStepIndex:flattenStepIndex color:(hasReachedEnd ? m_color : m_forwardPathColor)];
+
+                } else {
+                    [self addLineCreationParamsForStep:step stepBefore:stepBefore stepAfter:stepAfter flattenStepIndex:flattenStepIndex color:m_color];
+                }
+            }
+            else
+            {
+                if(isActiveStep)
+                {
+                    [self addLineCreationParamsForStep:step stepIndex:flattenStepIndex closestPointOnPath:closestPointOnRoute splitIndex:indexOfPathSegmentStartVertex];
+                }
+                else
+                {
+                    [self addLineCreationParamsForStep:step ndFlattenStepIndex:flattenStepIndex];
+                    
+                }
```

Comment:
Inconsistent formatting.

Reply:
Also inconsistent parameter naming. See `stepIndex` vs `ndFlattenStepIndex` for overloads of method `addLineCreationParamsForStep`.